### DISCUSSION TITLE
V3 - wire:navigation error on return

### DISCUSSION THREAD

**Discussion by tomphilpotts** [livewire/livewire#6030] [7. Bugs (archived)]
Hello,

I'm encountering an issue when using wire:navigate outside a Livewire component, specifically when I navigate back to a page that contains a Livewire component.

My setup involves a results page that uses a Livewire component for filtering. I have added wire:navigate to a link on a result card that navigates to a single result page. When I click the link, I am taken to the new page without a page reload, indicating wire:navigate works as expected. However, when I navigate back, I encounter the error: Uncaught TypeError: "Snapshot missing on Livewire component with id: " is not a constructor.

Interestingly, this issue occurs only when the Livewire component on the results page is updated (e.g., clicking a few filters), and then a result link is clicked. On navigating back, the error occurs. But if I navigate directly to a result (without updating any filters) and then navigate back, I do not encounter this issue.

This behavior suggests that the snapshot of the Livewire component state is not persisting when navigating back to the page. I believe this may be a potential bug that needs to be addressed.

Here's a screenshot of the error message I'm encountering:

![Screenshot 2023-07-27 at 12 46 51](https://github.com/livewire/livewire/assets/45283082/7256b0a6-96d7-44a4-b206-5484846becce)

I have seen that there is awareness of this issue, but not relating to this situation.

Thank tou

**Comment by joshhanley** [comment]
@tomphilpotts thanks for reporting! Any chance you could provide an example repo that demonstrates the issue with as little code as possible, so we can investigate? Thanks!

**Reply by tomphilpotts** [reply]
@joshhanley here is the repo
https://github.com/tomphilpotts/livewire-nav-test

**Reply by calebporzio** [reply]
Thanks for the reproduction. I fixed the original issue (try now after updating your composer.json livewire/livewire version to: 'dev-main' and running "composer update livewire/livewire")
